addon/ng2/utilities/ast-utils.ts

```diff
@@ -0,0 +1,54 @@
+import * as ts from 'typescript';
+import { InsertChange } from './change';
+
+/**
+* Find all nodes from the AST in the subtree of node of SyntaxKind kind.
+* @param node 
+* @param kind (a valid index of ts.SyntaxKind enum, eg ts.SyntaxKind.ImportDeclaration)
+* @return all nodes of kind kind, or [] if none is found
+*/
+export function findNodes (node: ts.Node, kind: number, arr: ts.Node[] = []): ts.Node[] {
```

Comment:
The `arr: ts.Node[]` parameter here is a little messy. Since the function returns an array already, why not make it also allocate its own array and accumulate values into that?